### ISSUE TITLE
Pull additional raindrop metadata

### ISSUE THREAD

**Issue by robertandrews** [kaiiiz/obsidian-raindrop-highlights-plugin#2]
- "created"
- "lastUpdate"
- "note"
- "type"
- "cover"

**Comment by kaiiiz** [comment]
There are three types of data that are pulled from Raindrop API.

1. Collections
2. Articles
3. Highlights

I'm not sure which type of data you're referring to. It's better to provide the API endpoint information and corresponding JSON path to prevent ambiguity.

**Comment by robertandrews** [comment]
I'm referring to "raindrops" (ie bookmarks). Not all raindrops (bookmarks) are articles (the "type" object also supports "video", "link" and maybe more).

So, the stuff available via the [/rest/v1/raindrops/{collectionId}](https://developer.raindrop.io/v1/raindrops/multiple) (multiple raindrops) endpoint. I think there's a bit more available than you're pulling...

```
{
"excerpt": "I was on a roll two weeks ago. I had managed, somehow, to find a prolonged and sustained flow. My work felt easier and more ...",
"note": "",
"type": "article",
"cover": ["https://blog.rescuetime.com/wp-content/uploads/2019/09/Unmotivated-motivation-wave.jpg"](https://blog.rescuetime.com/wp-content/uploads/2019/09/Unmotivated-motivation-wave.jpg),
"tags": [ ],
"removed": false,
"_id": 4281XXXXX,
"title": "A beautiful life is all about maintenance - RescueTime Blog",
-"collection": {
"$ref": "collections",
"$id": 2614XXXX,
"$db": ""
},
"link": ["https://blog.rescuetime.com/a-beautiful-life-is-all-about-maintenance/"](https://blog.rescuetime.com/a-beautiful-life-is-all-about-maintenance/),
"created": "2022-08-06T16:32:50.872Z",
"lastUpdate": "2022-08-06T20:08:05.544Z",
"important": false,
-"media": [
-{
"type": "image",
"link": ["https://blog.rescuetime.com/wp-content/uploads/2019/09/Unmotivated-motivation-wave.jpg"](https://blog.rescuetime.com/wp-content/uploads/2019/09/Unmotivated-motivation-wave.jpg)
},
-{
"type": "image",
"link": ["https://i0.wp.com/blog.rescuetime.com/wp-content/uploads/2019/09/Unmotivated-motivation-wave.jpg?resize=1000%2C486&ssl=1"](https://i0.wp.com/blog.rescuetime.com/wp-content/uploads/2019/09/Unmotivated-motivation-wave.jpg?resize=1000%2C486&ssl=1)
}
],
-"user": {
"$ref": "users",
"$id": 88XXXX,
"$db": ""
},
```

It may also be useful to make the Collection data title available as a template tag, but that would probably require a secondary request to /rest/v1/collections/{id}

**Comment by kaiiiz** [comment]
Yes, there are some fields that have not been pulled from the Raindrop article. I think it's fairly easy to add support to these fields.

The only thing I don't understand is the "note" field for each raindrop. Do you know how to add a note to a bookmark? It seems like this note field is different from the note field for highlight (annotation), and I haven't found a way to add a note to a bookmark on the Raindrop web page.

In addition, the latest collection data is saved in settings.json, so I think it's possible to support render Collection data in the template.

**Comment by robertandrews** [comment]
> The only thing I don't understand is the "note" field for each raindrop. Do you know how to add a note to a bookmark? It seems like this note field is different from the note field for highlight (annotation), and I haven't found a way to add a note to a bookmark on the Raindrop web page.

Not Pro so I don’t know if that setting shows there. I can’t see any such option in the UI. And it’s not excerpt either, is it? Possible to set note on create and see if one shows anywhere, perhaps.

**Comment by kaiiiz** [comment]
Excerpt is automatically generated by Raindrop when the new bookmark is created and can be edited under the title of the edit pane. No, there is no note field on the interface while creating a new bookmark.

**Comment by kaiiiz** [comment]
This feature has been implemented in v0.0.10 release.

**Comment by robertandrews** [comment]
I see it. Speedy coding. Thanks.
